sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -0,0 +1,182 @@
+        self.credential = credential
+        self.options = options
+
+        guard let endpointUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL.")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        let client = try AzureCommunicationChatClient(
+            endpoint: endpointUrl,
+            authPolicy: authPolicy,
+            withOptions: options
+        )
+
+        self.service = client.azureCommunicationChatService
+    }
+
+    // MARK: Public Methods
+
+    /// Create a ChatThreadClient for the ChatThread with id threadId.
+    /// - Parameters:
+    ///   - forChatThread: The threadId.
+    public func createClient(forChatThread threadId: String) throws -> ChatThreadClient {
+        return try ChatThreadClient(
+            endpoint: endpoint,
+            credential: credential,
+            threadId: threadId,
+            withOptions: options
+        )
+    }
+
+    /// Create a new ChatThread and return the ChatThreadClient for it.
+    /// - Parameters:
+    ///   - chatThread: Request for creating a chat thread with the topic and members to add.
+    ///   - withOptions: Create chat thread options.
+    ///   - completionHandler: A completion handler that receives a ChatThreadClient on success.
+    public func create(
+        chatThread: CreateChatThreadRequest,
+        withOptions options: AzureCommunicationChatService.CreateChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<ChatThreadClient>
+    ) {
+        service.create(chatThread: chatThread, withOptions: options) { result, httpResponse in
+            do {
+                switch result {
+                case let .success(chatThread):
+                    guard let threadId = chatThread.id else {
+                        throw AzureError.service("Service response does not contain ThreadId.")
+                    }
+
+                    // Create and return a ChatThreadClient for the new ChatThread
+                    let chatThreadClient = try ChatThreadClient(
+                        endpoint: self.endpoint,
+                        credential: self.credential,
+                        threadId: threadId,
+                        withOptions: self.options
+                    )
+
+                    completionHandler(.success(chatThreadClient), httpResponse)
+
+                case let .failure(error):
+                    throw error
+                }
+            } catch {
+                let azureError = AzureError.client("Failed to create ChatThread.", error)
+                completionHandler(.failure(azureError), httpResponse)
+            }
+        }
+    }
+
+    /// Get the ChatThread with given id.
+    /// - Parameters:
+    ///   - withId: The chat thread id.
+    ///   - withOptions: Get chat thread options.
+    ///   - completionHandler: A completion handler that receives the chat thread on success.
```

Comment:
Again, the labels here should be the internal labels, not the external one, per XCode's convention.

Reply:
Fixed the docs, thanks!